portal/src/main/webapp/js/src/study-view/component/PieChart.js

```diff
@@ -960,21 +967,22 @@ var PieChart = function(){
 
     //Called when the number of label biggen than 6, used by addPieLabels()
     function labelFunction() {
-        var _tableDiv = '';
+        var _tableDiv = [];
+        var _tableDivStr = '';
         var _labelLT = labelWLT; //label length threshold
         var _labelHeaderLT = labelHeaderWLT; //label header length threshold
 
-        _tableDiv += '<table';
+        _tableDiv.push('<table');
 
         if(category[0] === 'w1') {
             _labelLT = labelLT;
             _labelHeaderLT = labelHeaderLT;
-            _tableDiv += ' class="smallTable" ';
+            _tableDiv.push(' class="smallTable" ');
         }else{
-            _tableDiv += ' class="regularTable" ';
+            _tableDiv.push(' class="regularTable" ');
         }
 
-        _tableDiv += '><thead><th>'+selectedAttrDisplay+'</th><th>#</th><th>Freq</th></thead><tbody>';
+        _tableDiv.push('><thead><th>'+selectedAttrDisplay+'</th><th>#</th><th>Freq</th></thead><tbody>');
```

Comment:
hm...that is a pity that jquery would be so slow...but does it have a dramatic impact here? This part of the code does not seem like a place that should have big impact in the performance (i.e. a small table is being rendered here)....do we really need this kind of optimization here?